working/macros/feature-specification.md

```diff
@@ -1092,21 +1092,13 @@ it specifies through parameter types:
 
 *   If the parameter type is `bool`, `double`, `int`, `Null`, `num`, `String`,
     `List`, `Set`, or `Map`, (or the nullable forms of any of those), then the
-    argument expression must be a boolean, number, null, string, list, set, or
-    map literal.
-
-    * Number literals may be negated.
-    * String literals may not contain any interpolation, but may be adjacent
-      strings, and may be raw strings.
-    * List, Set and Map literals may only contain entries matching any of the
-      supported argument types. If the parameter type specifies a generic type
-      argument, it must be one of the allowed parameter types or `Object`,
-      recursively. Note that `Object` is allowed in order to exclude null items,
-      but all the actual entries must be of one of the supported types.
-
-    **TODO**: Do we want to allow more complex expressions? Could we allow
-    constant expressions whose identifiers can be successfully resolved before
-    macro expansion (#1929)?
+    argument expression must be a constant expression containing only boolean,
+    number, null, string, list, set, or map literals.
+
+    * Any type arguments (inferred or explicit) must be one of the allowed
+      parameter types or `Object`, recursively. Note that `Object` is allowed
```

Comment:
Probably need to call out that `Object?` is allowed too since it doesn't fall out from the previous bullet point.

Reply:
I moved this around a bit, added `Object` and `dynamic` explicitly above.